massa-execution-worker/src/speculative_ledger.rs

```diff
@@ -535,6 +540,17 @@ impl SpeculativeLedger {
             )));
         }
 
+        // check if caller doesn't have already too many data entries
+        if let Some(keys) = self.get_keys(addr, &[]) {
+            if keys.len() >= self.max_datastore_entry_count as usize {
```

Comment:
superiority comparison should be strict, because at line 536 we make sure that the entry already exists

Reply:
The entry is not existing yet and so if we do a strict superiority comparison we will endup with a datastore with too much obejcts